lib/chef/node.rb

```diff
@@ -196,51 +198,25 @@ def [](attrib)
     # might be missing
     def normal
       attributes.top_level_breadcrumb = nil
-      attributes.set_unless_value_present = false
       attributes.normal
     end
 
-    alias_method :set, :normal
-
-    # Set a normal attribute of this node, auto-vivifying any mashes that are
-    # missing, but if the final value already exists, don't set it
-    def normal_unless
-      attributes.top_level_breadcrumb = nil
-      attributes.set_unless_value_present = true
-      attributes.normal
+    def set
+      Chef.log_deprecation("node.set is deprecated and will be removed in Chef 14, please use node.default/node.override (or node.normal only if you really need persistence)")
```

Comment:
(actually rust does it one better with the short error message with the code in it, and then the ability to expand the code into a wall of text, and then sometimes with a shout out to a URL -- i'm insanely jealous of them, but googling 'chef attributes normal persistence' in this case looks entirely reasonable)